xtesting/check/check.go

```diff
@@ -0,0 +1,41 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package check provides a set of functions that can be used to check for errors in tests.
+package check
```

Comment:
I didn't get the difference between this and directly using `require.*`.

Reply:
`require.*` function signatures are different so you can't unify them under the single type definition. For example `NoError(t TestingT, err error, msgAndArgs ...any)` has three arguments while `EqualError(t TestingT, theError error, errString string, msgAndArgs ...any)` has four. `require.Regexp` doesn't make sense without check for error existence.

Reply:
Ah ok I see, it is for the table tests.